lang/cs/Org.Apache.REEF.Network/Elastic/Operators/Logical/ElasticOperator.cs

```diff
@@ -0,0 +1,552 @@
+                .BindIntNamedParam<OperatorParameters.Checkpointing>("" + (int)_checkpointLevel)
+                .Build();
+
+            operatorConf = Configurations.Merge(
+                operatorConf,
+                operatorBuilderWithTaskConf,
+                operatorBuilderWithTaskAndPhysicalConf,
+                Configurations.Merge(_configurations));
+
+            Stage.Context.SerializeOperatorConfiguration(ref serializedOperatorsConfs, operatorConf);
+        }
+
+        /// <summary>
+        /// Returns whether a failure should be propagated to downstream operators or not.
+        /// </summary>
+        /// <returns>True if the failure has to be sent downstream</returns>
+        protected virtual bool PropagateFailureDownstream()
+        {
+            return true;
+        }
+
+        /// <summary>
+        /// Operator specific logic for reacting when a task message is received.
+        /// </summary>
+        /// <param name="message">Incoming message from a task</param>
+        /// <param name="returnMessages">Zero or more reply messages for the task</param>
+        /// <returns>True if the operator has reacted to the task message</returns>
+        protected virtual bool ReactOnTaskMessage(ITaskMessage message, ref List<IElasticDriverMessage> returnMessages)
+        {
+            return false;
+        }
+
+        /// <summary>
+        /// Logs the current operator state.
+        /// </summary>
+        protected virtual void LogOperatorState()
+        {
+            string intro = $"State for Operator {OperatorType.ToString()} in Stage {Stage.StageName}:\n";
+            string topologyState = $"Topology:\n{_topology.LogTopologyState()}";
+            string failureMachineState = "Failure State: " + _failureMachine.State.FailureState +
+                    "\nFailure(s) Reported: " + _failureMachine.NumOfFailedDataPoints;
+
+            Log.Log(Level.Info, intro + topologyState + failureMachineState);
+        }
+
+        /// <summary>
+        /// Log the final internal statistics of the operator.
+        /// </summary>
+        protected virtual string LogInternalStatistics()
+        {
+            return _topology.LogFinalStatistics();
+        }
+
+        /// <summary>
+        /// Binding from logical to physical operator.
+        /// </summary>
+        /// <returns>The physcal operator configuration</returns>
+        protected abstract IConfiguration PhysicalOperatorConfiguration();
+
+        private ITopology GetTopology(TopologyType topologyType)
+        {
+            ITopology topology;
+
+            switch (topologyType)
+            {
+                case TopologyType.Flat:
+                    topology = new FlatTopology(MasterId);
+                    break;
+
+                default:
+                    throw new ArgumentException(
+                        nameof(topologyType),
+                        $"Topology type {topologyType} not supported by {OperatorType.ToString()}.");
```

Comment:
explicit `.ToString()` call is redundant

Reply:
Fixed.